clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -833,21 +835,23 @@ PointerVariableConstraint::mkString(Constraints &CS,
     if (!ForItype && InferredGenericIndex == -1 && isVoidPtr())
       K = Atom::A_Wild;
 
-    if (PrevArr && ArrSizes.at(TypeIdx).first != O_SizedArray && !EmittedName) {
-      EmittedName = true;
+    // In a case like `_Ptr<TYPE> p[2]`, the ` p[2]` needs to end up _after_ the
+    // `>`, so we need to push the ` p[2]` onto EndStrs _before_ the code below
+    // pushes the `>`. In general, before we visit a checked pointer level (not
+    // a checked array level), we need to transfer any pending array levels and
+    // emit the name (if applicable).
```

Comment:
This says the same thing twice (in different ways). You can choose one and it will be consistent with the rest of the comments here.

Reply:
> This says the same thing twice (in different ways). You can choose one

Not really: it is giving an example and then giving a general principle motivated by the example.  We do this all the time in our documentation.  If I take away either one of the two parts, I think it significantly increases the learning curve for a reader to understand this (still somewhat subtle) code.  If you still disagree, let's get another opinion from John.

> it will be consistent with the rest of the comments here.

Inasmuch as there's an inconsistency, I'm not convinced that it's a problem.  If it is, the right solution might be to add _more_ comments to the existing code, but I think that's beyond my responsibility in this PR.  Again, we could get another opinion if needed.

Reply:
We decided [not to change the text of this comment](https://github.com/correctcomputation/checkedc-clang/pull/714#pullrequestreview-771916818) and [not to start collapsing comments by default yet](https://correct-computation.slack.com/archives/G01GKGKHMFD/p1633534068026900?thread_ts=1633460151.019500&cid=G01GKGKHMFD).